libs/store_helper/user/ebpf_registry_helper.cpp

```diff
@@ -0,0 +1,246 @@
+    if (root_key == nullptr) {
+        return EBPF_INVALID_ARGUMENT;
+    }
+
+    return _EBPF_RESULT(RegCreateKeyEx(root_key, sub_key, 0, nullptr, 0, flags, nullptr, key, nullptr));
+}
+
+_Must_inspect_result_ ebpf_result_t
+open_registry_key(
+    ebpf_store_key_t root_key, _In_opt_z_ const wchar_t* sub_key, uint32_t flags, _Out_ ebpf_store_key_t* key)
+{
+    ebpf_assert(root_key != nullptr);
+    _Analysis_assume_(root_key != nullptr);
+
+    return _EBPF_RESULT(RegOpenKeyEx(root_key, sub_key, 0, flags, key));
+}
+
+_Must_inspect_result_ ebpf_result_t
+delete_registry_key(ebpf_store_key_t root_key, _In_z_ const wchar_t* sub_key)
+{
+    return _EBPF_RESULT(RegDeleteKeyEx(root_key, sub_key, 0, 0));
+}
+
+_Must_inspect_result_ ebpf_result_t
+delete_registry_tree(ebpf_store_key_t root_key, _In_opt_z_ const wchar_t* sub_key)
+{
+    return _EBPF_RESULT(RegDeleteTree(root_key, sub_key));
+}
+
+_Must_inspect_result_ ebpf_result_t
+create_registry_key_ansi(
+    ebpf_store_key_t root_key, _In_z_ const char* sub_key, uint32_t flags, _Out_ ebpf_store_key_t* key)
+{
+    ebpf_result_t result;
+    try {
+        auto wide_string = _get_wstring_from_string(sub_key);
+        result = create_registry_key(root_key, wide_string.c_str(), flags, key);
+    } catch (...) {
+        result = EBPF_NO_MEMORY;
+    }
+
+    return result;
+}
+
+_Must_inspect_result_ ebpf_result_t
+read_registry_value_string(
+    ebpf_store_key_t key, _In_z_ const wchar_t* value_name, _Outptr_result_maybenull_ wchar_t** value)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    unsigned long type = REG_SZ;
+    unsigned long value_size = 0;
+    wchar_t* string_value = nullptr;
+
+    if (value == nullptr) {
+        return EBPF_INVALID_ARGUMENT;
+    }
+
+    *value = nullptr;
+    result = _EBPF_RESULT(RegQueryValueEx(key, value_name, 0, &type, nullptr, &value_size));
+    if (result != EBPF_SUCCESS || type != REG_SZ) {
+        if (type != REG_SZ) {
+            result = EBPF_INVALID_ARGUMENT;
+        }
+        return result;
+    }
+
+    string_value = (wchar_t*)ebpf_allocate((value_size + sizeof(wchar_t)));
+    if (string_value == nullptr) {
+        return EBPF_NO_MEMORY;
+    }
+
+    memset(string_value, 0, value_size + sizeof(wchar_t));
+    result = _EBPF_RESULT(RegQueryValueEx(key, value_name, 0, &type, (PBYTE)string_value, &value_size));
+    if (result != EBPF_SUCCESS) {
+        goto Exit;
+    }
+
+    *value = string_value;
+    string_value = nullptr;
+
+Exit:
+    if (string_value) {
+        ebpf_free(string_value);
+    }
+    return result;
+}
+
+_Must_inspect_result_ ebpf_result_t
+read_registry_value_dword(ebpf_store_key_t key, _In_z_ const wchar_t* value_name, _Out_ uint32_t* value)
+{
+    unsigned long type = REG_QWORD;
+    unsigned long value_size = sizeof(uint32_t);
+    return _EBPF_RESULT(RegQueryValueEx(key, value_name, 0, &type, (PBYTE)value, &value_size));
+}
+
+_Must_inspect_result_ ebpf_result_t
+read_registry_value_binary(
```

Comment:
nit: for any non-static apis that are part of ebpf (e.g., returning ebpf_result_t) we try to prefix them with `ebpf_` to avoid risk of name collisions with functions in the caller.

Reply:
Np normalizing this now.